class, 변수, 메서드를 가지고 실제로 생성하면 인스턴스가 만들어진다~

A a = new A()

이 객체 안에서 기본적으로 세팅이 됐으면 좋겠음 => 생성자

## 생성자
어떤 일을 시작하기 전에 준비를 하는 것

객체가 만들어 졌을 때 그 객체의 기본값을 주는 역할

객체 생성, 인스턴스가 만들어질 때 그 인스턴스의 기본값을 주는 것

사람에게는 이름과 나이가 있다

첫째 name age

둘째 name age

셋째 name age

### 생성자 만드는 법 (생성자는 여러개 만들 수 o, 각각의 인자값이 달라야..)
```
public class Person {
 String name;
 int age;

public Person() {
 this.name="사람"
 this.age=1;
 }
 
 public Person( String name, int age){
  this.name = name;
  this.age = l
 } 

 public void toPrint(){
 system.out.println (this.name+"님의 나이"
 }
}
```
- person 객체 생성
- 변수 생성
- 인자 없는 생성자의 경우, this는 나의..라고 해석, 인자 없는 객체를 생성, 기본적으로 name에 사람 age에 1 
- 인자 없는 생성자 호출하는 법: Person a = new Person();
- 인자 있는 생성자의 경우, 두번째 person이라는 생성자에는 name, age라는 인자를 받는다
- 인자 있는 생성자 호출하는 법 :  Person P = new Person("이",1)
- this.name=name 해석.. 인자로 받은 name을 this.name에 집어넣겠다!
- this.age=1 해석.. 인자로 받은 age, 1값을 this.age에 집어넣겠다!



예시..
```
public static void main(String[] args)
{
Person p1 = new Person();
p1.toPrint();

Person p2 = new Person(“이홍직”,40);
p2,toPrint();
}
```

p1 name 사람 age 1
p2 name 이홍직 age 40

## 예외

예기치 못한 오류에 대한 처리 try, catch finally

try: 기본

catch : 예외 만나면 예외 처리

finally : 써도 되고 안 써도 됨, 예외 처리 후 무조건 실행됨

```
 public static void main(String[] args){
  try {
   코드 실행
  }
  catch(ArithmeticException e){
   산술 계산 오류 발생시 실행
  }
  catch(Exception e){
   모든 오류 발생시 실행
  }
  finally {
    모든 오류처리가 종료된 후 실행
  }
 }
```

### 예외 처리
```
public static void main (String[] args) {
int a = 10;
int b = 2;
try {
 System.out.println(“Start”);
 System.out.println(“결과값은”+(a/b));
 System.out.println(“End”);
}
catch(ArithmeticException e){
 System.out.println(“산술오류 발생”);
}
catch(Exception e){
 System.out.println(“오류 발생”)
}
finally{
 System.out.println(“프로그램 종료”)
}
```
